packages/grid/x-grid-modules/src/hooks/root/useEvents.ts

```diff
@@ -72,8 +71,8 @@ export function useEvents(
         const id = getIdFromRowElem(rowEl);
         const rowModel = apiRef.current.getRowFromId(id);
         const rowIndex = apiRef.current.getRowIndexFromId(id);
-        const field = getDataFromElem(cellEl, 'field');
-        const value = getDataFromElem(cellEl, 'value');
+        const field = cellEl.getAttribute('data-field');
+        const value = cellEl.getAttribute('data-value');
```

Comment:
IMHO, an unnecessary abstraction solves no bug, makes it harder to understand what's going on.